ASM/c/dungeon_info.c

```diff
@@ -293,12 +295,30 @@ void draw_dungeon_info(z64_disp_buf_t* db) {
 
         left += icon_size + padding;
 
-        // Draw dungeon names
-
+        // Draw the list of dungeons.
+        // Pre completed dungeons are grayed and crossed out.
         for (int i = 0; i < rows; i++) {
+            gDPPipeSync(db->p++);
             dungeon_entry_t* d = &(dungeons[i]);
+            int empty = CFG_DUNGEON_PRECOMPLETED[d->index];
             int top = start_top + ((icon_size + padding) * i) + 1;
-            text_print_size(db, d->name, left, top, font_width, font_height);
+            if (empty == 1) {
+                gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0x7F);
+                int sizeRectangle = text_print_size(db, d->name, left, top, font_width, font_height) - left;
+                gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0xBF);
+                gDPSetCombineMode(db->p++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+                gSPTextureRectangle(db->p++,
+                        left<<2, (top + 5) <<2,
+                        (left + sizeRectangle)<<2, ((top + 5) + 1)<<2,
+                        0,
+                        0, 0,
+                        1<<10, 1<<10);
```

Comment:
```suggestion
                        left * 4, (top + 5) * 4,
                        (left + sizeRectangle) * 4, ((top + 5) + 1) * 4,
                        0,
                        0, 0,
                        1024, 1024);
```
I don't like using bit shifts in code, makes it harder to read for an operation the compiler can optimize. You also shouldn't do this with signed integers in the first place. The compiler can't optimize signed multiplication by a power of two to a bit shift due to the second reason you shouldn't do this; two's complement representation of negative values.

Also shifting 1 outside of a macro instead of using a constant is unnecessary obfuscation.